v2/pkg/options/assetserver/middleware.go

```diff
@@ -0,0 +1,20 @@
+package assetserver
+
+import (
+	"net/http"
+)
+
+// Middleware defines a HTTP middleware that can be applied to the AssetServer.
+// The handler passed as next is the next handler in the chain. One can decide to call the next handler
+// or implement a specialized handling.
+type Middleware func(next http.Handler) http.Handler
+
+// ChainMiddleware allows chaining multiple middlewares to one middleware.
+func ChainMiddleware(middleware ...Middleware) Middleware {
```

Comment:
FYI, I have not found myself using this, but I can't say it wouldn't be useful to others (or even myself in the future). Just wanted to let you know that I was able to do what I needed with a single middleware. Since I control all of the middleware handlers, I can just multiplex them from my single middleware function.

Reply:
Thanks for letting me know. I didn't want to have an array of middleware in the API, so I thought this would round it up with this for people that want to chain multiple middlewares.